### PR TITLE
Add createEvent that returns the created Event

### PR DESCRIPTION
**Pull request by dbismut**
**What**:

Exposes a utility to create events that can be thrown by `fireEvent`. This is useful in order to have access to the source event attributes see #257

**Why**:

Some components might rely on event attributes created when the event object is created (i.e. `new Event({...eventAttributes})`).

**How**:

As suggested by @kentcdodds in #257:

```js
import {createEvent} from 'dom-testing-library'

const myEvent = createEvent.click(node, eventAttributes)
fireEvent(node, myEvent)
```

However since the event constructor depends on `window` which itself depends on the node, `createEvent` also requires the node to be passed as an argument (not very elegant I admit).

**Checklist**:

- [ ] Documentation added to the [docs site](https://github.com/alexkrolick/testing-library-docs)
- [x] Typescript definitions updated
- [ ] Tests: since `fireEvent` now uses `createEvent` I believe successful tests on `fireEvent` cover `createEvent`
- [x] Ready to be merged 

Pre-commit hook threw an error on Typescript definitions but I'm really not sure what's wrong, so I committed with the `--no-verify` flag.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,8 @@
     "test:update": "npm test -- --updateSnapshot --coverage",
     "validate": "kcd-scripts validate",
     "setup": "npm install && npm run validate -s",
-    "dtslint": "dtslint typings"
+    "dtslint": "dtslint typings",
+    "precommit": "kcd-scripts pre-commit"
```
Comment (kentcdodds) [inline package.json]: Could you revert these changes?

Comment (dbismut) [reply package.json]: Humpf sorry. I was trying to see what was wrong with my commit on typescript definitions...

**Comment by kentcdodds** [comment]
Could you make a pull request to the docs site to add documentation?

**Comment by kentcdodds** [comment]
@allcontributors[bot] please add @dbismut for code

**Comment by dbismut** [comment]
Yes I will PR the docs (although my English might be shaky)! Is it necessary to add createEvent as an export to react-testing-library?

**Comment by kentcdodds** [comment]
react-testing-library re-experts everything from dom-testing-library so we're good there 👍
